src/lightning/fabric/wrappers.py

```diff
@@ -179,6 +179,31 @@ def __getattr__(self, item: Any) -> Any:
             self._validate_method_access(item, attr)
             return attr
 
+    def __setattr__(self, name: str, value: Any) -> None:
+        caller_func = inspect.stack()[1].function
+        if caller_func == "__init__":
+            super().__setattr__(name, value)
```

Comment:
This is a neat trick, but I'm wondering if it's save. In theory, and it is not so far fetched IMO, one could define a Trainer like this:

```py
class MyTrainer:
    def __init__(self, model, ...):
        self.fabric = Fabric(...)
        self.model = fabric.setup(model)
        self.model.trainer = self
        ...
        
```
I'm not saying the above is a good practice or anything. But in this case I believe we would see the Trainer's init in `inspect.stack()[1].function`, but the intention is to have this code only triggered for FabricModule's init.

Reply:
I agree it is not far fetched. I have resorted to defining a flag variable at class level, to help us keep track whether the instance is initialized.